src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -0,0 +1,3926 @@
+                // (8,16): error CS0417: 'A': cannot provide arguments when creating an instance of a variable type
+                //         A v1 = new(1);
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "new(1)").WithArguments("A").WithLocation(8, 16),
+                // (9,16): error CS0417: 'B': cannot provide arguments when creating an instance of a variable type
+                //         B v2 = new(2);
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "new(2)").WithArguments("B").WithLocation(9, 16),
+                // (10,16): error CS0304: Cannot create an instance of the variable type 'C' because it does not have the new() constraint
+                //         C v3 = new();
+                Diagnostic(ErrorCode.ERR_NoNewTyvar, "new()").WithArguments("C").WithLocation(10, 16)
+                );
+        }
+
+        [Fact]
+        public void TestTypeParameter()
+        {
+            var source = @"
+using System;
+
+struct S
+{
+    static void M1<T>() where T : struct
+    {
+        Console.Write((T)new());
+    }
+    static void M2<T>() where T : new()
+    {
+        Console.Write((T)new());
+    }
+
+    public static void Main()
+    {
+        M1<S>();
+        M2<S>();
+    }
+}
+";
+
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "SS");
+        }
+
+        [Fact]
+        public void TestTypeParameterInitializer()
+        {
+            var source = @"
+using System;
+
+class C
+{
+    public int field;
+
+    static void M1<T>() where T : C, new()
+    {
+        Console.Write(((T)new(){ field = 42 }).field);
+    }
+
+    public static void Main()
+    {
+        M1<C>();
+    }
+}
+";
+
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "42");
+        }
+
+        [Fact]
+        public void TestInitializer_ErrorCase()
+        {
+            var source = @"
+class C
+{
+    public static void Main()
+    {
+        string x = new() { Length = 5 };
+    }
+}
+";
+
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics(
+                // (8,28): error CS0200: Property or indexer 'string.Length' cannot be assigned to -- it is read only
+                //         string x = new() { Length = 5 };
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "Length").WithArguments("string.Length").WithLocation(6, 28),
+                // (8,20): error CS1729: 'string' does not contain a constructor that takes 0 arguments
+                //         string x = new() { Length = 5 };
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "new() { Length = 5 }").WithArguments("string", "0").WithLocation(6, 20)
+                );
+        }
+
+        [Fact]
+        public void TestImplicitConversion()
```

Comment:
We generally prefer to use `Theory`s and `InlineData` for these types of tests, with interpolated strings, to avoid manual string replacement.

Reply:
Or just splitting them up. You already have tests for inaccessible constructors, I'm not sure what extra this tests over other existing tests